local/ob-tmate/ob-tmate.el

```diff
@@ -379,49 +381,50 @@ Argument OB-SESSION: the current ob-tmate session."
               (setenv "TERMINFO" osx-terminfo)
             )
             ))
-    ;;unset tmux env so this can be run from within a tmux session without complaint
-    (setenv "TMUX")
-
-  (start-process-shell-command
-   (concat session-name "-tmate-process")
-   (concat "**" session-name "-tmate-process**")
-   (concat "nohup tmate"
-           ;; " -n " "init"; session-window
-           ;; " -F -v"
-           " -v"
-           ;; " -d -v"
-           " -S " session-socket
-           " new-session"
-           " -d -P -F '#{tmate_ssh}:#{tmate_web}'"
-           " -s " session-name
-           " -c " session-dir
-           " -n 0 " ; This is window 0
-           " bash -c \"" ; begin command
-           "( " ; start wrap to display errors
-           "echo Waiting for tmate..." ; wait for tmate ready
-           " && "
-           ;; wait for tmate to be fully ready
-           "tmate wait tmate-ready "
-           " && "
-           "echo '\nShare this only with people you trust:'"
-           " && "
-           "tmate display -p '#{tmate_ssh} # " session-name "'"
-           " && "
-           "tmate display -p '#{tmate_web} # " session-name "'"
-           " && "
-           "echo '\nShare this read only connection otherwise:'"
-           " && "
-           "tmate display -p '#{tmate_ssh_ro} # " session-name "'"
-           " && "
-           "tmate display -p '#{tmate_web_ro} # " session-name "'"
-           " && "
-           ;; Let folks know what to do with this
-           "echo '\nShare the above connection with a friend and check huemacs'"
-           " ) 2>&1" ; end wrap to display errors
-           " && "
-           "read X" ; need to hit enter to continue
-           "\"" ; end command
-           ))
+  (let ((process-environment (cl-copy-list process-environment)))
```

Comment:
We do this twice.... once for when we create new tmate session/sockets and again for when we execute tmate strings.